src/clarity-angular/emphasis/alert/alerts.spec.ts

```diff
@@ -0,0 +1,259 @@
+import {Alert} from "./alert";
+
+describe("Alerts component", function() {
+   describe("Typescript API", function() {
+      let component: Alerts;
+      let service: MultiAlertService;
+      let queryList: QueryList<Alert>;
+
+      beforeEach(function() {
+         service = new MultiAlertService();
+
+         TestBed.configureTestingModule({
+             imports: [ClrEmphasisModule]
+         });
+
+         let alertFixture = TestBed.createComponent(Alert);
+         this.alert = alertFixture.componentInstance;
+
+         let anotherAlertFixture = TestBed.createComponent(Alert);
+         this.anotherAlert = anotherAlertFixture.componentInstance;
+
+         queryList = new QueryList<Alert>();
+         queryList.reset([
+            this.alert,
+            this.anotherAlert
+         ]);
+         service.manage(queryList);
+         component = new Alerts(service);
+      });
+
+      it("knows the current alert", function() {
+         expect(component.currentAlert).toEqual(this.alert);
+      });
+   });
+
+   describe("Template API", function() {
+
+      let fixture: ComponentFixture<any>;
+
+      beforeEach(function() {
+         this.create = <T>(componentType: Type<T>) => {
+            TestBed.configureTestingModule({
+               imports: [ClrEmphasisModule],
+               declarations: [componentType]
+            });
+
+            fixture = TestBed.createComponent(componentType);
+            // fixture.detectChanges();
+         };
+      });
+
+      afterEach(() => {
+         fixture.destroy();
+      });
+
+      it("offers two way binding on the alert index", function() {
+         this.create(TestComponent);
+         const testComponent = fixture.componentInstance;
+         fixture.detectChanges();
+         testComponent.currentAlertIndex = 0;
+         fixture.detectChanges();
+         expect(testComponent.alertsInstance.currentAlertIndex).toEqual(0);
+         testComponent.alertsInstance.currentAlertIndex = 1;
+         fixture.detectChanges();
+         expect(testComponent.currentAlertIndex).toEqual(1);
+      });
+
+      // TODO - Fix
+      it("offers two way binding on the alert instance", fakeAsync(function() {
+          this.create(TestAlertInstance);
+          fixture.detectChanges();
+
+          let clarityComponent = fixture.componentInstance.alertsInstance;
+          let testComponent = fixture.componentInstance;
+
+          expect(testComponent.alertInstances.length).toBe(2);
+          expect(testComponent.alertInstances.first).toBeTruthy();
+          expect(testComponent.alertInstances.last).toBeTruthy();
+
+          testComponent.currentAlert = testComponent.alertInstances.first;
+          fixture.detectChanges();
+          expect(clarityComponent.currentAlert).toEqual(testComponent.currentAlert);
+
+          clarityComponent.currentAlert = testComponent.alertInstances.last;
+          // TODO: figure out why this makes the test hang
```

Comment:
Still working on addressing the last part of this test but would appreciate an initial review so I can address any comments while working on this.